### PR TITLE
TST, CI: reproduce wheels 32-bit setup

### LINKED ISSUE #11095: BUG: find_peaks_cwt test failures in 32-bit Linux wheels
The combination of Linux + 32-bit + somewhat-older versions of NumPy seems to cause the failures.

[Example Travis test matrix](https://travis-ci.org/MacPython/scipy-wheels/builds/614182055?utm_source=github_status&utm_medium=notification)

```
=================================== FAILURES ===================================
____________________ TestFindPeaksCwt.test_find_peaks_exact ____________________
[gw0] linux -- Python 3.5.7 /venv/bin/python

self = <scipy.signal.tests.test_peak_finding.TestFindPeaksCwt object at 0xe8fd7dcc>

    def test_find_peaks_exact(self):
        """
        Generate a series of gaussians and attempt to find the peak locations.
        """
        sigmas = [5.0, 3.0, 10.0, 20.0, 10.0, 50.0]
        num_points = 500
        test_data, act_locs = _gen_gaussians_even(sigmas, num_points)
        widths = np.arange(0.1, max(sigmas))
        found_locs = find_peaks_cwt(test_data, widths, gap_thresh=2, min_snr=0,
                                         min_length=None)
        np.testing.assert_array_equal(found_locs, act_locs,
>                       "Found maximum locations did not equal those expected")
E       AssertionError: 
E       Arrays are not equal
E       Found maximum locations did not equal those expected
E       (mismatch 100.0%)
E        x: array([ 72, 143, 215, 286, 358, 429], dtype=int32)
E        y: array([ 71, 142, 214, 285, 357, 428])

act_locs   = array([ 71, 142, 214, 285, 357, 428])
found_locs = array([ 72, 143, 215, 286, 358, 429], dtype=int32)
num_points = 500
self       = <scipy.signal.tests.test_peak_finding.TestFindPeaksCwt object at 0xe8fd7dcc>
sigmas     = [5.0, 3.0, 10.0, 20.0, 10.0, 50.0]
test_data  = array([  1.50549685e-32,   2.11937869e-32,   2.98119127e-32,
         4.19009319e-32,   5.88450706e-32,   8.25750906e-....75099657e-01,
         1.66027806e-01,   1.57300074e-01,   1.48911963e-01,
         1.40858421e-01,   1.33133885e-01])
widths     = array([  0.1,   1.1,   2.1,   3.1,   4.1,   5.1,   6.1,   7.1,   8.1,
         9.1,  10.1,  11.1,  12.1,  13.1,  14.1,... 35.1,
        36.1,  37.1,  38.1,  39.1,  40.1,  41.1,  42.1,  43.1,  44.1,
        45.1,  46.1,  47.1,  48.1,  49.1])

/venv/lib/python3.5/site-packages/scipy/signal/tests/test_peak_finding.py:789: AssertionError
__________________ TestFindPeaksCwt.test_find_peaks_withnoise __________________
[gw0] linux -- Python 3.5.7 /venv/bin/python

self = <scipy.signal.tests.test_peak_finding.TestFindPeaksCwt object at 0xe8fdb16c>

    def test_find_peaks_withnoise(self):
        """
        Verify that peak locations are (approximately) found
        for a series of gaussians with added noise.
        """
        sigmas = [5.0, 3.0, 10.0, 20.0, 10.0, 50.0]
        num_points = 500
        test_data, act_locs = _gen_gaussians_even(sigmas, num_points)
        widths = np.arange(0.1, max(sigmas))
        noise_amp = 0.07
        np.random.seed(18181911)
        test_data += (np.random.rand(num_points) - 0.5)*(2*noise_amp)
        found_locs = find_peaks_cwt(test_data, widths, min_length=15,
                                         gap_thresh=1, min_snr=noise_amp / 5)
    
        np.testing.assert_equal(len(found_locs), len(act_locs), 'Different number' +
                                'of peaks found than expected')
        diffs = np.abs(found_locs - act_locs)
        max_diffs = np.array(sigmas) / 5
        np.testing.assert_array_less(diffs, max_diffs, 'Maximum location differed' +
>                                    'by more than %s' % (max_diffs))
E       AssertionError: 
E       Arrays are not less-ordered
E       Maximum location differedby more than [  1.    0.6   2.    4.    2.   10. ]
E       (mismatch 33.33333333333333%)
E        x: array([1, 1, 1, 3, 1, 0])
E        y: array([  1. ,   0.6,   2. ,   4. ,   2. ,  10. ])

act_locs   = array([ 71, 142, 214, 285, 357, 428])
diffs      = array([1, 1, 1, 3, 1, 0])
found_locs = array([ 72, 143, 215, 288, 358, 428], dtype=int32)
max_diffs  = array([  1. ,   0.6,   2. ,   4. ,   2. ,  10. ])
noise_amp  = 0.07
num_points = 500
self       = <scipy.signal.tests.test_peak_finding.TestFindPeaksCwt object at 0xe8fdb16c>
sigmas     = [5.0, 3.0, 10.0, 20.0, 10.0, 50.0]
test_data  = array([  6.78864387e-02,  -2.00643540e-02,  -3.27414303e-02,
         1.19094918e-02,   5.52614013e-02,   6.31457252e-....28751325e-01,
         2.10417219e-01,   1.06335978e-01,   1.70557941e-01,
         1.30800895e-01,   2.02734269e-01])
widths     = array([  0.1,   1.1,   2.1,   3.1,   4.1,   5.1,   6.1,   7.1,   8.1,
         9.1,  10.1,  11.1,  12.1,  13.1,  14.1,... 35.1,
        36.1,  37.1,  38.1,  39.1,  40.1,  41.1,  42.1,  43.1,  44.1,
        45.1,  46.1,  47.1,  48.1,  49.1])

/venv/lib/python3.5/site-packages/scipy/signal/tests/test_peak_finding.py:811: AssertionError
```

### PR DESCRIPTION
**Pull request by tylerjereddy**
* pin 32-bit Linux azure CI NumPy version
to minimum supported to produce better
match to wheels repo (should reveal two
`signal` test failures)

* failures are reported in https://github.com/scipy/scipy/issues/11095 ; I'm working on the fix for them, but probably good to have main repo with less dissonance with wheels repo

* example wheels fail: https://travis-ci.org/github/MacPython/scipy-wheels/builds/687683011

Fixes #11095

### CODE REVIEW

**Comment by rgommers** [comment]
Useful for fixing these failures to pin to 1.14.5. It does bring up the question if we shouldn't have two jobs; this just flips from newest to oldest supported NumPy so we may miss other things.

**Comment by tylerjereddy** [comment]
I thought of that. At the moment only one case blocks wheels releases though. I can just add another job I guess if there isn't a lot of  resistance to that.

**Comment by rgommers** [comment]
I'm not sure, happy to merge as is (once it's green) and worry about it later. Fixing things up and branching the release seems more important.

**Comment by tylerjereddy** [comment]
This patch solves one of the two failures:

```diff
--- a/scipy/signal/wavelets.py
+++ b/scipy/signal/wavelets.py
@@ -298,6 +298,7 @@ def ricker(points, a):
     """
     A = 2 / (np.sqrt(3 * a) * (np.pi**0.25))
     wsq = a**2
+    points = np.ceil(points)
     vec = np.arange(0, points) - (points - 1.0) / 2
     xsq = vec**2
     mod = (1 - xsq / wsq)
```

Something about the ceiling/arange size calculation is finnicky for 32-bit + old NumPy.

I see floating points values like 81.00000004 and then ceiling produces 82, while I see 81.0/81.0 for input/ceiling in the working case (newer NumPy/64-bit).

Worth noting that `points` is supposed to be `int` according to the docstring, but almost never is... Seems to be `float64` typically.

The remaining failure is even more subtle now:

```
scipy/signal/tests/test_peak_finding.py:786: in test_find_peaks_exact
    "Found maximum locations did not equal those expected")
E   AssertionError:
E   Arrays are not equal
E   Found maximum locations did not equal those expected
E   (mismatch 16.66666666666667%)
E    x: array([ 71, 142, 214, 285, 357, 429], dtype=int32)
E    y: array([ 71, 142, 214, 285, 357, 428])
```

If I replace `np.ceil(points)` with `np.int64(points)` the both tests pass on 32-bit, but then another `signal` test has an issue.

**Comment by tylerjereddy** [comment]
Ah, never mind, it is just the `int64` coercion that solves both, but that causes another fail. Surprisingly messy situation...

**Comment by tylerjereddy** [comment]
@rgommers Ok, I've added a commit that allows the full `signal` test suite to pass locally in a 32-bit Docker container with `numpy==1.14.5`.

I don't claim that this is a beautiful fix--it is based on empirical debug printing of the floating point ceiling values between newer and older NumPy in the `signal` test suite to develop a "functional shim" for old NumPy.

**Review by larsoner — APPROVED** [review]
Tough to diagnose (sounds like) but the fixes seem like reasonable shims. This gets things green again, +1 for merge from me

**Comment by tylerjereddy** [comment]
Ok, core dev +1 to merge and Ralf seemed on board once the bug was fixed. I need the wheels CI green so I will merge.

Someone can follow-up with another 32-bit CI entry using latest NumPy if we think that is justified.

**Pull request merged**
